### ISSUE TITLE
Separate the error and warning popups from the documentation popups in Intellisense

### ISSUE THREAD

**Issue by aderchox** [microsoft/vscode#118105]
Currently, there are [some](https://code.visualstudio.com/docs/editor/intellisense#_customizing-intellisense) options available for customizing how the intellisense popups behave, including when it should appear, the delay before its appearance and a few more. However, I've not found any option to turn on or off the intellisense error- & warning-popups and the documentation popups separately. (I've searched using Google, Stackoverflow and Github, so if it's already been suggested, then I've not been lucky enough to find them).

Here are also two screenshots to explain it in action:
![2021-03-04_13-21-13](https://user-images.githubusercontent.com/52244768/109947821-fcc8a280-7cee-11eb-8f5b-1c2c6d597c12.png)
![2021-03-04_13-36-27](https://user-images.githubusercontent.com/52244768/109947847-03571a00-7cef-11eb-9b67-3a8cc8f6f30f.png)

Thanks.

**Comment by aderchox** [comment]
@IllusionMH How is it confusing? Tell me what explanation you need and I will edit the issue to add it. IMO, It's very simple. The popup consists of three parts, one is for "Peek Problem/Quick Fix", another one is for warning or errors, and the top one allows reading some documentation. I've seen many people turning off the intellisense popups **only** because it covers(hides) so many lines, but also disabling it altogether using the `"editor.hover.enabled": false` will be at the expense of losing all the warning and error hints as well.

**Comment by IllusionMH** [comment]
Your request is request is clear - if it was totally unclear then someone would post comment with explicit request to provide more details.

In this request you want to hide info on hover which is main reason for hover popup - provide info about methods/variables etc.

In the same time there is dedicated command `>Go to Next Problem ...` <kbd>Alt</kbd>+<kbd>F8</kbd> (on Windows) which iterates over all errors in document and has different view for error only.
![image](https://user-images.githubusercontent.com/1312662/109951607-92136900-7ce6-11eb-881d-869c37041033.png)

**Comment by aderchox** [comment]
Even if the initial purpose of it has been showing the content in the top part(the docs), I (subjectively) believe that still the bottom parts are **far more frequently** required.

**Comment by IllusionMH** [comment]
No need to argue with me - my opinions on priority ("...main reason for hover popup...") and frequency (when I need to see error long descriptions are rarely on my way, but I need to see both type info and error message at the same time) are also subjective and I'm not team member so I have no effect on decision.

My as described above - my reaction is because there is already dedicated view for problems as title requests (no Quick Fixes though, rarely useful for me) and both info and error are useful during fixed. YMMW

**Comment by yume-chan** [comment]
For some well-documented languages/libraries, the documentation portion can be very very very long, thus requiring multiple scrolls to get to the error message.

My suggestion is at least scrolling documentation and error message separately, each has its own maximum height.

```
-------------------
doc               ↑
doc               |
doc               |
doc               ↓
-------------------
error             ↑
error             |
error             |
error             ↓
-------------------
```

**Comment by aderchox** [comment]
@yume-chan Yes, the second screenshot I've put is exactly an example of this, I had to scroll down so much until I got to the end of the docs(API note) of the type "BigDecimal" and then I was able to see the error.

 What's interesting is that I think even now the errors and docs are being loaded separately, because when the error is shown, there's usually a tiny delay before the docs load and get added to the pop up. Of course it's worth noting that this may also become misleading to some people, as it may sound as if the docs _covered/replaced_ the error message, while indeed the error has just gone way down in the pop up.

**Comment by RReverser** [comment]
> thus requiring multiple scrolls to get to the error message

The worst part is that when scrolling quickly with a touchpad, you can often scroll past the end of both docs and error message, and then instead of stopping in the end, it starts scrolling document underneath.

So user needs to go back to the original error location, hover again, wait for the tooltip, and scroll more carefully and slowly to actually get to the error message.

**Comment by lf-** [comment]
This may duplicate https://github.com/microsoft/vscode/issues/73120

**Comment by aderchox** [comment]
> This may partially duplicate #73120

Thanks for the reference, yes that's similar, but IMO mine might be asking for a bit more, since I've recommended adding an option in the settings explicitly to be able to customize this in different ways. But yes, totally the same issue.

**Comment by lf-** [comment]
> > This may partially duplicate #73120
> 
> Thanks for the reference, yes that's similar, but IMO mine might be asking for a bit more, since I've recommended adding an option in the settings explicitly to be able to customize this in different ways. But yes, totally the same issue.

Agreed! I just was looking and saw both issues so wanted to make sure they're all linked to make it easier for the next person :)
